### ISSUE TITLE
In Android 10 not working

### ISSUE THREAD

**Issue by jaydeep2158** [DreaminginCodeZH/MaterialRatingBar#66]
Hello sir,
 I have to implement your code but in android 10 not working. When I have run application throw exception like Fatal Exception: android.view.InflateException: Binary XML file Error inflating class me.zhanghai.android.materialratingbar.MaterialRatingBar

Thanks
Jaydeep Dhamecha

**Comment by zhanghai** [comment]
Are you testing on Samsung Android 10? Are you using the latest version (1.4 0 or 1.3.3) of this library?

**Comment by jaydeep2158** [comment]
Yes, I have to use Samsung Android 10 and use latest version of library is 1.4.0

**Comment by zhanghai** [comment]
Can you provide more detailed stacktrace? There should be some more lines following it. Ideally something that contains the `materialratingbar` package will help a lot.

**Comment by jaydeep2158** [comment]
**1)InflateException**
->Fatal Exception: android.view.InflateException: Binary XML file line #120 in com.kraven:layout/test: Binary XML file line #120 in com.test:layout/test: Error inflating class me.zhanghai.android.materialratingbar.MaterialRatingBar

**2)InvocationTargetException**
->Caused by java.lang.reflect.InvocationTargetException
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance + 343(Constructor.java:343)
       at android.view.LayoutInflater.createView + 854(LayoutInflater.java:854)
       at android.view.LayoutInflater.createViewFromTag + 1006(LayoutInflater.java:1006)
       at android.view.LayoutInflater.createViewFromTag + 961(LayoutInflater.java:961)
       at android.view.LayoutInflater.rInflate + 1123(LayoutInflater.java:1123)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.inflate + 682(LayoutInflater.java:682)
       at android.view.LayoutInflater.inflate + 534(LayoutInflater.java:534)
       at com.kraven.ui.base.BaseFragment.onCreateView + 66(BaseFragment.kt:66)
       at androidx.fragment.app.Fragment.performCreateView + 2600(Fragment.java:2600)
       at androidx.fragment.app.FragmentManagerImpl.moveToState + 881(FragmentManagerImpl.java:881)
       at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState + 1238(FragmentManagerImpl.java:1238)
       at androidx.fragment.app.FragmentManagerImpl.moveToState + 1303(FragmentManagerImpl.java:1303)
       at androidx.fragment.app.BackStackRecord.executeOps + 439(BackStackRecord.java:439)
       at androidx.fragment.app.FragmentManagerImpl.executeOps + 2079(FragmentManagerImpl.java:2079)
       at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether + 1869(FragmentManagerImpl.java:1869)
       at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute + 1824(FragmentManagerImpl.java:1824)
       at androidx.fragment.app.FragmentManagerImpl.execPendingActions + 1727(FragmentManagerImpl.java:1727)
       at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange + 2663(FragmentManagerImpl.java:2663)
       at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated + 2613(FragmentManagerImpl.java:2613)
       at androidx.fragment.app.FragmentController.dispatchActivityCreated + 246(FragmentController.java:246)
       at androidx.fragment.app.FragmentActivity.onStart + 542(FragmentActivity.java:542)
       at androidx.appcompat.app.AppCompatActivity.onStart + 201(AppCompatActivity.java:201)
       at android.app.Instrumentation.callActivityOnStart + 1433(Instrumentation.java:1433)
       at android.app.Activity.performStart + 8004(Activity.java:8004)
       at android.app.ActivityThread.handleStartActivity + 3471(ActivityThread.java:3471)
       at android.app.servertransaction.TransactionExecutor.performLifecycleSequence + 221(TransactionExecutor.java:221)
       at android.app.servertransaction.TransactionExecutor.cycleToPath + 201(TransactionExecutor.java:201)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState + 173(TransactionExecutor.java:173)
       at android.app.servertransaction.TransactionExecutor.execute + 97(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage + 2146(ActivityThread.java:2146)
       at android.os.Handler.dispatchMessage + 107(Handler.java:107)
       at android.os.Looper.loop + 237(Looper.java:237)
       at android.app.ActivityThread.main + 7762(ActivityThread.java:7762)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run + 493(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main + 1047(ZygoteInit.java:1047)


**3)NullPointerException**
->Caused by java.lang.NullPointerException: Attempt to read from field 'android.content.res.ColorStateList me.zhanghai.android.materialratingbar.MaterialRatingBar$TintInfo.mProgressTintList' on a null object reference
       at me.zhanghai.android.materialratingbar.MaterialRatingBar.getSupportProgressTintList + 315(MaterialRatingBar.java:315)
       at me.zhanghai.android.materialratingbar.MaterialRatingBar.getProgressTintList + 156(MaterialRatingBar.java:156)
       at android.widget.AbsSeekBar.<init> + 245(AbsSeekBar.java:245)
       at android.widget.RatingBar.<init> + 92(RatingBar.java:92)
       at android.widget.RatingBar.<init> + 88(RatingBar.java:88)
       at android.widget.RatingBar.<init> + 124(RatingBar.java:124)
       at me.zhanghai.android.materialratingbar.MaterialRatingBar.<init> + 44(MaterialRatingBar.java:44)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance + 343(Constructor.java:343)
       at android.view.LayoutInflater.createView + 854(LayoutInflater.java:854)
       at android.view.LayoutInflater.createViewFromTag + 1006(LayoutInflater.java:1006)
       at android.view.LayoutInflater.createViewFromTag + 961(LayoutInflater.java:961)
       at android.view.LayoutInflater.rInflate + 1123(LayoutInflater.java:1123)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.rInflate + 1126(LayoutInflater.java:1126)
       at android.view.LayoutInflater.rInflateChildren + 1084(LayoutInflater.java:1084)
       at android.view.LayoutInflater.inflate + 682(LayoutInflater.java:682)
       at android.view.LayoutInflater.inflate + 534(LayoutInflater.java:534)
       at com.kraven.ui.base.BaseFragment.onCreateView + 66(BaseFragment.kt:66)
       at androidx.fragment.app.Fragment.performCreateView + 2600(Fragment.java:2600)
       at androidx.fragment.app.FragmentManagerImpl.moveToState + 881(FragmentManagerImpl.java:881)
       at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState + 1238(FragmentManagerImpl.java:1238)
       at androidx.fragment.app.FragmentManagerImpl.moveToState + 1303(FragmentManagerImpl.java:1303)
       at androidx.fragment.app.BackStackRecord.executeOps + 439(BackStackRecord.java:439)
       at androidx.fragment.app.FragmentManagerImpl.executeOps + 2079(FragmentManagerImpl.java:2079)
       at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether + 1869(FragmentManagerImpl.java:1869)
       at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute + 1824(FragmentManagerImpl.java:1824)
       at androidx.fragment.app.FragmentManagerImpl.execPendingActions + 1727(FragmentManagerImpl.java:1727)
       at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange + 2663(FragmentManagerImpl.java:2663)
       at androidx.fragment.app.FragmentManagerImpl.dispatchActivityCreated + 2613(FragmentManagerImpl.java:2613)
       at androidx.fragment.app.FragmentController.dispatchActivityCreated + 246(FragmentController.java:246)
       at androidx.fragment.app.FragmentActivity.onStart + 542(FragmentActivity.java:542)
       at androidx.appcompat.app.AppCompatActivity.onStart + 201(AppCompatActivity.java:201)
       at android.app.Instrumentation.callActivityOnStart + 1433(Instrumentation.java:1433)
       at android.app.Activity.performStart + 8004(Activity.java:8004)
       at android.app.ActivityThread.handleStartActivity + 3471(ActivityThread.java:3471)
       at android.app.servertransaction.TransactionExecutor.performLifecycleSequence + 221(TransactionExecutor.java:221)
       at android.app.servertransaction.TransactionExecutor.cycleToPath + 201(TransactionExecutor.java:201)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState + 173(TransactionExecutor.java:173)
       at android.app.servertransaction.TransactionExecutor.execute + 97(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage + 2146(ActivityThread.java:2146)
       at android.os.Handler.dispatchMessage + 107(Handler.java:107)
       at android.os.Looper.loop + 237(Looper.java:237)
       at android.app.ActivityThread.main + 7762(ActivityThread.java:7762)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run + 493(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main + 1047(ZygoteInit.java:1047)

**Comment by ynaseem1990** [comment]
yes  i got the same issue in S10 plus also,

**Comment by JeremiahStephenson** [comment]
I was getting the same error with 1.3.2 but updating to 1.4.0 fixed it for me.

**Comment by JeremiahStephenson** [comment]
The unfortunate thing though is the app I work on broke when Samsung released the Android 10 update for Galaxy S10s and it was because of this library. The fallout has hit me hard because of this so I would actually recommend ditching this library and going back to the regular framework RatingBar.

**Comment by zhanghai** [comment]
This is the same stacktrace as in #63 and updating to `1.3.3` or `1.4.0` will include the fix (1005cd3). Please confirm that the crash and stacktrace is actually reproduced with these two versions, because the fix was specifically working around this NPE.

```
NullPointerException
->Caused by java.lang.NullPointerException: Attempt to read from field 'android.content.res.ColorStateList me.zhanghai.android.materialratingbar.MaterialRatingBar$TintInfo.mProgressTintList' on a null object reference
at me.zhanghai.android.materialratingbar.MaterialRatingBar.getSupportProgressTintList + 315(MaterialRatingBar.java:315)
at me.zhanghai.android.materialratingbar.MaterialRatingBar.getProgressTintList + 156(MaterialRatingBar.java:156)
at android.widget.AbsSeekBar. + 245(AbsSeekBar.java:245)
at android.widget.RatingBar. + 92(RatingBar.java:92)
at android.widget.RatingBar. + 88(RatingBar.java:88)
at android.widget.RatingBar. + 124(RatingBar.java:124)
at me.zhanghai.android.materialratingbar.MaterialRatingBar. + 44(MaterialRatingBar.java:44)
at java.lang.reflect.Constructor.newInstance0(Constructor.java)
at java.lang.reflect.Constructor.newInstance + 343(Constructor.java:343)
at android.view.LayoutInflater.createView + 854(LayoutInflater.java:854)
```

**Comment by yccheok** [comment]
@jaydeep2158 

Are u using Samsung Galaxy S10 ?

**Comment by jaydeep2158** [comment]
> @jaydeep2158
> 
> Are u using Samsung Galaxy S10 ?

Yes

**Comment by jaydeep2158** [comment]
> @jaydeep2158
> 
> Are u using Samsung Galaxy S10 ?

But my issue is resolved when I used 1.4.0 library.

**Comment by yccheok** [comment]
Thanks. I just wanna to know, whether this issue is applied to all Android 10, or only Samsung Galaxy S10's Android 10.

May I know do you only encounter this issue, after you have upgrade Android 9 to Android 10, in Samsung Galaxy S10?

Recently, I encounter other type of crash issue (Not related to MaterialRatingBar library), which only happen in Samsung Galaxy S10 Android 10.
